lib/mix/lib/mix/tasks/escript.build.ex

```diff
@@ -108,7 +108,7 @@ defmodule Mix.Tasks.Escript.Build do
   defp escriptize(project, language, force, should_consolidate) do
     escript_opts = project[:escript] || []
 
-    script_name  = to_string(escript_opts[:name] || project[:app])
+    script_name  = Mix.Escript.escript_name(project)
```

Comment:
I don't think it's useful for escripts. `archive.build` works with non-mix projects according to its docs. Escripts require mix.exs, so it would be simpler to have only one place where the name can be changed – in mix.exs.